engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -7260,7 +7262,11 @@ object IR {
 
       /** @inheritdoc */
       override val location: Option[IdentifiedLocation] =
-        ast.location.map(IdentifiedLocation(_, ast.id))
+        try {
+          ast.location.map(IdentifiedLocation(_, ast.id))
+        } catch {
+          case _: ClassCastException => None
```

Comment:
```scala
at match {
  case ast: AST => Some(location)
  case _ => None
}
```
And we can remove `ast` function

Reply:
True, `match` is nicer: [9eece87](https://github.com/enso-org/enso/pull/3723/commits/9eece87dba167ac3aba11e29eb51bc3c68215b48)